core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/promisehandlers/StartWorkflowHandler.scala

```diff
@@ -4,9 +4,11 @@ import com.twitter.util.Future
 import edu.uci.ics.amber.engine.architecture.controller.ControllerAsyncRPCHandlerInitializer
 import edu.uci.ics.amber.engine.architecture.controller.promisehandlers.StartWorkflowHandler.StartWorkflow
 import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand
+import edu.uci.ics.texera.web.workflowruntimestate.WorkflowAggregatedState
+import edu.uci.ics.texera.web.workflowruntimestate.WorkflowAggregatedState.RUNNING
 
 object StartWorkflowHandler {
-  final case class StartWorkflow() extends ControlCommand[Unit]
+  final case class StartWorkflow() extends ControlCommand[WorkflowAggregatedState]
```

Comment:
I am hesitating to let the start workflow return the state. Here you are returning RUNNING directly without quering the actual workflow state. It is cleaner to return None and let other messages extract the state (e.g., query statistics) in a more aggregated way. If you want to make the change, at least you need to do the same for the Python side.

Reply:
Changed it to let it always return the state from the actual workflow state. Python side does not handle this message. Only the controller does.